awx/main/tasks.py

```diff
@@ -2322,8 +2295,7 @@ def pre_run_hook(self, instance, private_data_dir):
             os.mkdir(settings.PROJECTS_ROOT)
         self.acquire_lock(instance)
         self.original_branch = None
-        if (instance.scm_type == 'git' and instance.job_type == 'run' and instance.project and
-                instance.scm_branch != instance.project.scm_branch):
+        if instance.scm_type == 'git' and instance.branch_override:
```

Comment:
Explanation: this is changed due to introduction of new property `branch_override`, and it's also true that `branch_override=True` implies `instance.job_type == 'run'` so I removed that from the condition.